compiler/rustc_middle/src/ty/vtable.rs

```diff
@@ -57,7 +58,7 @@ pub(super) fn vtable_allocation_provider<'tcx>(
 
         tcx.vtable_entries(trait_ref)
     } else {
-        COMMON_VTABLE_ENTRIES
+        common_vtable_entries()
```

Comment:
I wonder if you could do `const COMMON_VTABLE_ENTRIES: &'tcx [VtblEntry<'tcx>] = common_vtable_entries(); COMMON_VTABLE_ENTRIES` here to guarantee that it gets const evaluated. (But perf will tell us if we need to try to be tricky).

Reply:
I don't think consts can reference arbitrary inner lifetimes, can they?

Reply:
Uh...probably not actually.

Reply:
Associated consts can, and inherent impl blocks can have associated consts

Reply:
neat, maybe I'll associate these consts with TyCtxt then